### PR TITLE
Fix South Carolina "senior exemption" calculation

### LINKED ISSUE #3394: Fix South Carolina "senior exemption" calculation
Using the 0.564.0 version, the following test fails because PEUS does not correctly offset the `sc_senior_exemption` amount by the `sc_retirement_deduction` amount as called for by the line q-1 and line q-2 worksheets in the [2021 SC income tax instruction booklet](https://dor.sc.gov/forms-site/Forms/IITPacket_2021.pdf#page=18).

```
- name: Tax unit with taxsimid 13715 in f21.its.csv and f21.ots.csv
  absolute_error_margin: 0.01
  period: 2021
  input:
    people:
      person1:
        age: 67
        employment_income: 27_010
        taxable_interest_income: 5_505
        taxable_private_pension_income: 6_000
        social_security_retirement: 11_500
        ssi: 0  # not in TAXSIM35
        state_supplement: 0  # not in TAXSIM35
        wic: 0  # not in TAXSIM35
      person2:
        age: 67
        employment_income: 30_010
        taxable_interest_income: 5_505
        taxable_private_pension_income: 6_000
        social_security_retirement: 11_500
        ssi: 0  # not in TAXSIM35
        state_supplement: 0  # not in TAXSIM35
        wic: 0  # not in TAXSIM35
    tax_units:
      tax_unit:
        members: [person1, person2]
        premium_tax_credit: 0  # not in TAXSIM35
        local_income_tax: 0  # not in TAXSIM35
        state_sales_tax: 0  # not in TAXSIM35
    spm_units:
      spm_unit:
        members: [person1, person2]
        snap: 0  # not in TAXSIM35
        tanf: 0  # not in TAXSIM35
    households:
      household:
        members: [person1, person2]
        state_fips: 45  # SC
  output:  # expected results from patched TAXSIM35 2023-12-07 version
    sc_retirement_deduction: 12_000
    sc_senior_exemption: 30_000 - 12_000
#
# TEST RESULTS FOR 0.564.0 VERSION:
# Test 'Tax unit with taxsimid 13715 in f21.its.csv and f21.ots.csv':
#   sc_senior_exemption@2021: [30000.] differs from 18000.0
#                             with an absolute margin [12000.] > 0.01
```

### PR DESCRIPTION
**Pull request by JudyZ13**
Fixes #3394

Co-authored-by: Pavel Makarchuk <PavelMakarchuk@users.noreply.github.com>

### Summary
:memo::test_tube::older_adult:

This pull request refactors the implementation of the South Carolina senior exemption, a tax benefit for elderly taxpayers, from a tax unit level variable to a person level variable. It adds two new Python files to define the person and tax unit level variables, updates the unit test cases, and adds a new integration test case. It also deletes some unused files related to other retirement income deductions in South Carolina. It adds an empty YAML file to store the changelog entries for the release process.

> _`sc_senior_exemption`_
> _Refactored to person level_
> _Autumn of life tax_

### Walkthrough
*  Refactor the South Carolina senior exemption from a tax unit level variable to a person level variable ([link](https://github.com/PolicyEngine/policyengine-us/pull/3396/files?diff=unified&w=0#diff-31f868403ed679b362f39bdbc0cc680079b72fc7735c70aa5674c6c7a562ad71), [link](https://github.com/PolicyEngine/policyengine-us/pull/3396/files?diff=unified&w=0#diff-74d86dd58851a53412222185f3f865ed9d51333ea6d0d56f8f938c7a8f822611), [link](https://github.com/PolicyEngine/policyengine-us/pull/3396/files?diff=unified&w=0#diff-6430136773d90f7ef1c5431187e69e27d298a9633edcc12c8f7d2f06555941dc), [link](https://github.com/PolicyEngine/policyengine-us/pull/3396/files?diff=unified&w=0#diff-cbbfeff642b2332c90b46a410dca1687d03b35e3085d18cec2a15e2d34b344c8), [link](https://github.com/PolicyEngine/policyengine-us/pull/3396/files?diff=unified&w=0#diff-03f10831ef541c2c19db305f16ccead8b6f01c42404c693d60848366d1303fb9), [link](https://github.com/PolicyEngine/policyengine-us/pull/3396/files?diff=unified&w=0#diff-58144be965cf46a4bf5d5a29be1625230c88de78401e693d8d0937c7eaf31a55R1-R13), [link](https://github.com/PolicyEngine/policyengine-us/pull/3396/files?diff=unified&w=0#diff-e3f28877d0945b4feefcc990aa1240fa9046a603eac62a0a72ad8ad124775efeR1-R39))
  - Delete the files `sc_military_retirement_income_deduction_head.py`, `sc_military_retirement_income_deduction_spouse.py`, `sc_retirement_income_deduction_head.py`, `sc_retirement_income_deduction_spouse.py`, and `sc_senior_exemption.py`, which defined the tax unit level variables for the senior exemption and its components ([link](https://github.com/PolicyEngine/policyengine-us/pull/3396/files?diff=unified&w=0#diff-31f868403ed679b362f39bdbc0cc680079b72fc7735c70aa5674c6c7a562ad71), [link](https://github.com/PolicyEngine/policyengine-us/pull/3396/files?diff=unified&w=0#diff-74d86dd58851a53412222185f3f865ed9d51333ea6d0d56f8f938c7a8f822611), [link](https://github.com/PolicyEngine/policyengine-us/pull/3396/files?diff=unified&w=0#diff-6430136773d90f7ef1c5431187e69e27d298a9633edcc12c8f7d2f06555941dc), [link](https://github.com/PolicyEngine/policyengine-us/pull/3396/files?diff=unified&w=0#diff-cbbfeff642b2332c90b46a410dca1687d03b35e3085d18cec2a15e2d34b344c8), [link](https://github.com/PolicyEngine/policyengine-us/pull/3396/files?diff=unified&w=0#diff-03f10831ef541c2c19db305f16ccead8b6f01c42404c693d60848366d1303fb9))
  - Add the files `sc_senior_exemption_person.py` and `sc_senior_exemption.py`, which define the person level variable `sc_senior_exemption_person` and the tax unit level variable `sc_senior_exemption`, respectively ([link](https://github.com/PolicyEngine/policyengine-us/pull/3396/files?diff=unified&w=0#diff-58144be965cf46a4bf5d5a29be1625230c88de78401e693d8d0937c7eaf31a55R1-R13), [link](https://github.com/PolicyEngine/policyengine-us/pull/3396/files?diff=unified&w=0#diff-e3f28877d0945b4feefcc990aa1240fa9046a603eac62a0a72ad8ad124775efeR1-R39))
  - Calculate the `sc_senior_exemption_person` based on the person's age, retirement income deductions, and filing status, and sum up the `sc_senior_exemption` for the tax unit ([link](https://github.com/PolicyEngine/policyengine-us/pull/3396/files?diff=unified&w=0#diff-58144be965cf46a4bf5d5a29be1625230c88de78401e693d8d0937c7eaf31a55R1-R13), [link](https://github.com/PolicyEngine/policyengine-us/pull/3396/files?diff=unified&w=0#diff-e3f28877d0945b4feefcc990aa1240fa9046a603eac62a0a72ad8ad124775efeR1-R39))
* Update the unit test cases for the South Carolina senior exemption to reflect the new input and output variables and values ([link](https://github.com/PolicyEngine/policyengine-us/pull/3396/files?diff=unified&w=0#diff-e62a4ad580bce673ebdcbe27c6015e7f9a1fabcc2aaecfc8af241691e7553970L6-R60))
  - Modify the file `sc_senior_exemption.yaml` in the `policyengine_us/tests/policy/baseline/gov/states/sc/tax/income/exemptions/senior` directory ([link](https://github.com/PolicyEngine/policyengine-us/pull/3396/files?diff=unified&w=0#diff-e62a4ad580bce673ebdcbe27c6015e7f9a1fabcc2aaecfc8af241691e7553970L6-R60))
  - Use the person level variables `sc_military_retirement_income_deduction_person` and `sc_retirement_income_deduction_person` as inputs, and the person level variable `sc_senior_exemption_person` as output ([link](https://github.com/PolicyEngine/policyengine-us/pull/3396/files?diff=unified&w=0#diff-e62a4ad580bce673ebdcbe27c6015e7f9a1fabcc2aaecfc8af241691e7553970L6-R60))
  - Adjust the expected values for the output variable based on the reference source ([link](https://github.com/PolicyEngine/policyengine-us/pull/3396/files?diff=unified&w=0#diff-e62a4ad580bce673ebdcbe27c6015e7f9a1fabcc2aaecfc8af241691e7553970L6-R60))
* Add a new integration test case for the South Carolina senior exemption ([link](https://github.com/PolicyEngine/policyengine-us/pull/3396/files?diff=unified&w=0#diff-d13c0843300f70000f352ecafc134f09e4e45f0e0df61d49ba3178ef581a5917R1-R41))
  - Add the file `intergration.yaml` in the `policyengine_us/tests/policy/baseline/gov/states/sc/tax/income/exemptions` directory ([link](https://github.com/PolicyEngine/policyengine-us/pull/3396/files?diff=unified&w=0#diff-d13c0843300f70000f352ecafc134f09e4e45f0e0df61d49ba3178ef581a5917R1-R41))
  - Use the tax unit level variable `sc_senior_exemption` as output, and compare it with the expected value from TAXSIM35 ([link](https://github.com/PolicyEngine/policyengine-us/pull/3396/files?diff=unified&w=0#diff-d13c0843300f70000f352ecafc134f09e4e45f0e0df61d49ba3178ef581a5917R1-R41))
* Add an empty YAML file to store the changelog entries for the pull request ([link](https://github.com/PolicyEngine/policyengine-us/pull/3396/files?diff=unified&w=0#diff-5885042762833a4aef51405a4036067e4e6aefa7a5091a2987d1d9d1fc2647e8R1))
  - Add the file `changelog_entry.yaml` in the root directory of the repository ([link](https://github.com/PolicyEngine/policyengine-us/pull/3396/files?diff=unified&w=0#diff-5885042762833a4aef51405a4036067e4e6aefa7a5091a2987d1d9d1fc2647e8R1))
  - Use the file to document the changes and features introduced by the pull request, following the format required by OpenFisca-BumpVersion ([link](https://github.com/PolicyEngine/policyengine-us/pull/3396/files?diff=unified&w=0#diff-5885042762833a4aef51405a4036067e4e6aefa7a5091a2987d1d9d1fc2647e8R1))

### CODE REVIEW

File: policyengine_us/tests/policy/baseline/gov/states/sc/tax/income/exemptions/sc_senior_exemption.yaml
Comment (PavelMakarchuk) [inline policyengine_us/tests/policy/baseline/gov/states/sc/tax/income/exemptions/sc_senior_exemption.yaml:1]: rename to sc_senior_exemption_person

File: changelog_entry.yaml [hunk 1]
```diff
@@ -0,0 +1 @@
+
```
Comment (PavelMakarchuk) [inline changelog_entry.yaml:1]: need changelog

File: changelog_entry.yaml
Comment (PavelMakarchuk) [inline changelog_entry.yaml:1]: run make format

File: policyengine_us/variables/gov/states/sc/tax/income/exemptions/senior/sc_senior_exemption_person.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+            [
+                "sc_retirement_deduction_indv",
+                "sc_military_deduction_indv",
+            ],
+        )
+
+        # Calculate senior exemption. The exemption can not be less than 0.
+        reduced_max_amount = max_(p.amount - total_deductions, 0)
+
+        return where(eligible, reduced_max_amount, 0)
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/sc/tax/income/exemptions/senior/sc_senior_exemption_person.py:40]:
```suggestion
        return eligible * reduced_max_amount
```

**Review by MaxGhenis — APPROVED** [review]

**Comment by MaxGhenis** [comment]
Just to summarize, looks like this also deletes redundant variables

**Pull request merged**